platform/platform-tests/testSrc/com/intellij/openapi/editor/impl/AccessibleTextTest.java

```diff
@@ -0,0 +1,164 @@
+    var stopOptions = CaretStopOptionsTransposed.DEFAULT_UNIX.toCaretStopOptions();
+    checkWordsWhenMovingBackward(stopOptions, false, "}", ";", ")", "!\"", "World", ",", "Hello", "\"", "(", "println",
+                                 ".", "out",
+                                 ".", "System",
+                                 "{", ")", "(", "printWelcomeMessage", "void", "static", "public");
+  }
+
+  public void testGetTextAtIndexReturnsCorrectWordWithWindowsDefaultStopOptionsWhenMovingForWard() {
+    var stopOptions = CaretStopOptionsTransposed.DEFAULT_WINDOWS.toCaretStopOptions();
+    checkWordsWhenMovingForward(stopOptions, false,
+                                "public", "static", "void", "printWelcomeMessage", "(", ")", "{", "\n",
+                                " ", "System", ".", "out", ".", "println", "(", "\"", "Hello", ",", "World", "!\"", ")", ";", "\n",
+                                " ", "}");
+  }
+
+  public void testGetTextAtIndexReturnsCorrectWordWithWindowsDefaultStopOptionsWhenMovingBackward() {
+    var stopOptions = CaretStopOptionsTransposed.DEFAULT_WINDOWS.toCaretStopOptions();
+    checkWordsWhenMovingBackward(stopOptions, false, "\n", "}", " ", "\n", ";", ")", "!\"", "World", ",", "Hello", "\"", "(", "println",
+                                 ".", "out",
+                                 ".", "System", " ",
+                                 "\n", "{", ")", "(", "printWelcomeMessage", "void", "static", "public");
+  }
+
+
+  private Editor initEditor(CaretStopOptions stopOptions, boolean isCamel) {
+    initText(text);
+
+    var editor = getEditor();
+    editor.getSettings().setCamelWords(isCamel);
+    EditorSettingsExternalizable.getInstance().setCaretStopOptions(stopOptions);
+    return editor;
+  }
+
+  private static void checkWords(Editor editor, Runnable action, String... expectedWords) {
+    var accessibleContext = editor.getContentComponent().getAccessibleContext();
+    var accessibleText = accessibleContext.getAccessibleText();
+    for (String expectedWord : expectedWords) {
+      action.run();
+      int offset = editor.getCaretModel().getOffset();
+      String actualWord = accessibleText.getAtIndex(AccessibleText.WORD, offset);
+      assertEquals(expectedWord, actualWord);
+    }
+  }
+
+  private void checkWords(Runnable initialAction, CaretStopOptions stopOptions, boolean isCamel, Runnable action, String... expectedWords) {
+    Editor editor = initEditor(stopOptions, isCamel);
+    initialAction.run();
+    checkWords(editor, action, expectedWords);
+  }
+
+  private void checkWordsWhenMovingForward(CaretStopOptions stopOptions, boolean isCamel, String... expectedWords) {
+    var textStartAction = new Runnable() {
+      @Override
+      public void run() {
+        textStart();
+      }
+    };
+    var nextWordAction = new Runnable() {
+      @Override
+      public void run() {
+        nextWord();
+      }
+    };
```

Comment:
Nitpick: consider using lambda expressions to make it shorter:
```suggestion
    Runnable textStartAction = () -> textStart();
    Runnable nextWordAction = () -> nextWord();
```